Bucketeer/Sources/Internal/Remote/ApiClientImpl.swift

```diff
@@ -156,7 +156,7 @@ final class ApiClientImpl: ApiClient {
                 }
                 do {
                     guard 200..<300 ~= urlResponse.statusCode else {
-                        let response: ErrorResponse? = try JSONDecoder().decode(ErrorResponse.self, from: data)
+                        let response: ErrorResponse? = try? JSONDecoder().decode(ErrorResponse.self, from: data)
```

Comment:
fixed: could not parser error when the body is empty or is not in expected JSON format

Reply:
@duyhungtnn, just confirming. Before this fixing, what error event was being reported?

Reply:
Hi @cre8ivejp ,before this. If the SDK fail to parse the response's body , it will report the error as `unknown` without checking the response's status code. But the error should depend on the response's status code. So that is not correct.

the updated test case related to this bug https://github.com/bucketeer-io/ios-client-sdk/pull/51/files#diff-b8becb5ad1412af7c03a24156dc3f8fa88fd400c78fbeb018cf35e82a8c0501dR829